pkg/service/dependencies.go

```diff
@@ -19,23 +21,31 @@ import (
 
 // Dependencies struct holds the reference to the Storage and the Manipulator interface implementations
 type Dependencies struct {
-	Storage     base.Storage
-	Manipulator Manipulator
+	Storage       base.Storage
+	Manipulator   Manipulator
+	MetricService metrics.MetricService
 }
 
 // NewDependencies constructs new Dependencies based on the config.DataSource().Kind
 // Currently, it supports only one Manipulator
-func NewDependencies() (*Dependencies, error) {
+func NewDependencies(registry *prometheus.Registry) (*Dependencies, error) {
+	var metricService metrics.MetricService
+	if regex.PrometheusMatcher.MatchString(config.MetricsSystem()) {
+		metricService = metrics.NewPrometheus(registry)
+	} else if regex.StatsdMatcher.MatchString(config.MetricsSystem()) {
+		metricService, _ = metrics.InitializeStatsdCollector(config.StatsdConfig())
+	}
+	deps := &Dependencies{Manipulator: NewManipulator(native.NewBildProcessor(), getDefaultParams(), metricService),
+		MetricService: metricService}
```

Comment:
We should have a `NoOpMetricsService` which only implements the interface but does nothing, it can be treated as default metricService, if the regex doesn't match, the `metricService` variable will have nil value and application will panic.